### DISCUSSION TITLE
Updating common styles in template css

### DISCUSSION THREAD

**Discussion by grahamford** [vmware/clarity#5830] [Q&A] [answered]
In Clarity 4 I could apply formatting in my templates like
```
p {
    margin-top: 0.2rem;
}
```

With version 5, this no longer works because the selector for the base styling from Clarity is now.
```
p:not([cds-text]) {
    ...
    margin-top: 1.2rem;
    ...
}
```

Am I missing something here, or do I need to add `:not([cds-text])` to everywhere I've used a css rule that is also in `@clr/angular`?

**Comment by mathisscott** [comment, marked as answer]
Broadly speaking, you should only need to do that for paragraphs and headers (`h1..h6`). And, yes, the path of least resistance is adding `:not([cds-text])` to your style declarations. Clarity is moving away from applying margin directly to HTML elements.

You know your application best, however. So it's possible you could use `!important` to force specificity if you needed to. Or you could apply a class name to your paragraphs and increase specificity that way. If this is a global requirement, you could also consider increasing specificity by indicating `body p` or some other known parent element.

**Reply by grahamford** [reply]
Thanks for the clarification @mathisscott 

The styling is often tweaks put into individual components so it's sometimes harder to find in order to update but we shall look at implementing this.
